src/app/Dashboard/Dashboard.tsx

```diff
@@ -36,21 +36,29 @@
  * SOFTWARE.
  */
 import * as React from 'react';
-import { Stack, StackItem } from '@patternfly/react-core';
+import { Card, CardActions, CardBody, CardHeader, Stack, StackItem, Text } from '@patternfly/react-core';
 import { useDispatch, useSelector } from 'react-redux';
-import { addCardIntent, deleteCardIntent } from '@app/Shared/Redux/DashboardConfigActions';
+import { deleteCardIntent } from '@app/Shared/Redux/DashboardConfigActions';
 import { TargetView } from '@app/TargetView/TargetView';
 import { AddCard } from './AddCard';
 import { DashboardCardActionMenu } from './DashboardCardActionMenu';
-import { AutomatedAnalysisCard } from './AutomatedAnalysis/AutomatedAnalysisCard';
+import { AutomatedAnalysisCardDescriptor } from './AutomatedAnalysis/AutomatedAnalysisCard';
 import { RootState, StateDispatch } from '@app/Shared/Redux/ReduxStore';
 
-export interface CardConfig {
+export interface DashboardCardDescriptor {
   title: string;
   description: string;
   descriptionFull: JSX.Element | string;
-  component: React.FunctionComponent;
-  props?: React.PropsWithChildren<any>;
+  component: React.FunctionComponent<any>;
+  propControls: PropControl[];
+}
+
+export interface PropControl {
+  name: string;
+  key: string;
+  description: string;
```

Comment:
Should this `description` prop be optional? Same for `defaultValue`?

Reply:
I don't think so, it seems like these should be required properties for the generated form.